tests/unit/htdocs/class/XoopsSecurityTest.php

```
@@ -0,0 +1,579 @@
+/**
+ * Testable version of XoopsSecurity with all real methods.
+ *
+ * Since the bootstrap stub only defines createToken(), we replicate
+ * the full real class methods here so they can be properly tested.
+ * This mirrors the source at class/xoopssecurity.php exactly.
+ */
+class TestableXoopsSecurity extends \XoopsSecurity
+{
+    public $errors = [];
+
+    public function check($clearIfValid = true, $token = false, $name = 'XOOPS_TOKEN')
```

PHPMD (2.15.0)

[error] The method check has a boolean flag argument $clearIfValid, which is a certain sign of a Single Responsibility Principle violation. (undefined)

(BooleanArgumentFlag)